src/index.ts

```diff
@@ -702,9 +720,10 @@ class ThreadPool {
       });
     }
 
-    // If there is a task queue, there's no point in looking for an available
-    // Worker thread. Add this task to the queue, if possible.
-    if (this.taskQueue.size > 0) {
+    // If there is a task queue, or there is unacceptable CPU load,
+    // there's no point in looking for an available Worker thread.
+    // Add this task to the queue, if possible.
+    if (this.taskQueue.size > 0 || !this._acceptableCpuLoad()) {
```

Comment:
If we hit this condition, I think should probably start a timer that checks whether CPU load has become acceptable or not, otherwise the task will hang around indefinitely.

Reply:
Yeah, that's where I was leaning but wanted to see if you ended up in the same place ;-) This is exactly why I skipped the check on the worker available event above.